From 68319e7f0755a7d36ed1544214dceae0afa05b76 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Thu, 25 Mar 2010 22:15:58 -0700
Subject: [PATCH 09/14] XQuartz: Workaround weird key data reported on some layouts

This should make 'Unicode Hex Input' work as an input layout.

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
Acked-by: Adam Jackson <ajax@redhat.com>
(cherry picked from commit 339207be6f184cc783076fc7e2cc12f92f57f2ba)
---
 hw/xquartz/quartzKeyboard.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git xorg-server-1.7.6/hw/xquartz/quartzKeyboard.c xorg-server-1.7.6/hw/xquartz/quartzKeyboard.c
index c9ef7cc..a4a0b08 100644
--- xorg-server-1.7.6/hw/xquartz/quartzKeyboard.c
+++ xorg-server-1.7.6/hw/xquartz/quartzKeyboard.c
@@ -735,7 +735,10 @@ Bool QuartzReadSystemKeymap(darwinKeyboardInfo *info) {
                     if (err != noErr) continue;
                 }
 
-                if (len > 0 && s[0] != 0x0010) {
+                /* Not sure why 0x0010 is there.
+                 * 0x0000 - <rdar://problem/7793566> 'Unicode Hex Input' ...
+                 */
+                if (len > 0 && s[0] != 0x0010 && s[0] != 0x0000) {
                     k[j] = ucs2keysym (s[0]);
                     if (dead_key_state != 0) k[j] = make_dead_key (k[j]);
                 }
-- 
1.7.0.2

